cunumeric/module.py

```diff
@@ -205,16 +205,11 @@ def choose(a, choices, out=None, mode="raise"):
     return array.choose(choices=choices, out=out, mode=mode, stacklevel=2)
 
 
-def diag_helper(a, offset=0, axes=None, extract=True):
-    array = ndarray.convert_to_cunumeric_ndarray(a)
-    return array.diag_helper(offset=offset, axes=axes, extract=extract)
-
-
 @copy_docstring(np.diag)
 def diag(v, k=0):
     array = ndarray.convert_to_cunumeric_ndarray(v)
     if array._thunk.scalar:
```

Comment:
This is too strict. `.scalar` is defined as "backed by a Future and size == 1`, which can still be a valid 1d or 2d array. You should skip this check altogether.

Reply:
should it be:?
```
  if array.ndim==0:
        raise ValueError("Input must be 1- or 2-d")
```

NumPy will return `ValueError: Input must be 1- or 2-d.` for :

```
  b=3
  print (np.diag(b))
```

Reply:
yeah that sounds right